pkg/api/validation/validation.go

```diff
@@ -756,21 +778,21 @@ func ValidateLimitRange(limitRange *api.LimitRange) errs.ValidationErrorList {
 	if len(limitRange.Name) == 0 {
 		allErrs = append(allErrs, errs.NewFieldRequired("name", limitRange.Name))
 	} else if !util.IsDNSSubdomain(limitRange.Name) {
-		allErrs = append(allErrs, errs.NewFieldInvalid("name", limitRange.Name, ""))
+		allErrs = append(allErrs, errs.NewFieldInvalid("name", limitRange.Name, DnsSubdomainErrorMsg))
 	}
 	if len(limitRange.Namespace) == 0 {
 		allErrs = append(allErrs, errs.NewFieldRequired("namespace", limitRange.Namespace))
 	} else if !util.IsDNSSubdomain(limitRange.Namespace) {
-		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", limitRange.Namespace, ""))
+		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", limitRange.Namespace, DnsSubdomainErrorMsg))
 	}
 	// ensure resource names are properly qualified per docs/resources.md
 	for i := range limitRange.Spec.Limits {
 		limit := limitRange.Spec.Limits[i]
 		for k := range limit.Max {
-			allErrs = append(allErrs, validateResourceName(string(k))...)
+			allErrs = append(allErrs, validateResourceName(string(k), string(limit.Type))...)
```

Comment:
I think you want to pass fmt.Sprintf("spec.limits[%d].max", i) here (and ".min" below) or even better:  fmt.Sprintf("spec.limits[%d].max[%s]", i, k)